File: src/protocol/java/packet/Formats.java
/*
 * Filename: Formats.java
 * FullPath: apps/CWSP-reborn/src/protocol/java/packet/Formats.java
 * Change date and time: 16.42.00_10.07.2026
 * Reason for changes: Transport/format name constants used in diagnostics.
 */
package space.u2re.cwsp.protocol.packet;

/** Transport and format name constants (wire-stable). */
public final class Formats {

    public static final String WS_PATH = "/ws";
    public static final String SOCKETIO_PATH = "/socket.io";

    public static final String TRANSPORT_WS = "ws";
    public static final String TRANSPORT_SOCKETIO = "socketio";
    public static final String TRANSPORT_BRIDGE = "bridge";
    public static final String TRANSPORT_HTTP = "http";

    // Preference order per network.mdc Transport Preference section.
    public static final String[] PREFERENCE = {"ws", "http", "tcp", "socketio"};
    public static final String[] FALLBACK = {"ws", "bridge", "socketio"};

    private Formats() {
    }
}

File: src/protocol/node/packet/Formats.ts
/*
 * Filename: Formats.ts
 * FullPath: apps/CWSP-reborn/src/protocol/node/packet/Formats.ts
 * Change date and time: 17.20.00_10.07.2026
 * Reason for changes: Thin NodeJS facade re-exporting the canonical wire normalization entrypoint. | Alias migration 17.20.00_10.07.2026: long relative cwsp-shared imports -> @fest-lib/cwsp-shared/v2/*.
 *
 * NOTE: Re-export only. `normalizePacketForWire` is the canonical pre-send sanitizer from
 * `@fest-lib/cwsp-shared` v2; the NodeJS endpoint must not duplicate sanitization.
 */

export {
    normalizePacketForWire,
    normalizePacket,
} from "@fest-lib/cwsp-shared/v2/normalize.ts";
